libera/src/main/groovy/com/connexta/libera/Project.groovy

```diff
@@ -0,0 +1,137 @@
+     *                    "<project>[:<releaseVersion>[:<nextVersion>]]" format
+     * @param nextProject reference to the next project to release. {@link #NO_OP_PROJECT} must
+     *                    be used if this project is the last one to be released.
+     */
+    protected Project(Libero libero, String projectInfo, Project nextProject) {
+        def strings = projectInfo.tokenize(":")
+
+        this.libero = libero
+        this.projectName = strings[0]
+        this.releaseVersion = strings[1]
+        this.nextVersion = strings[2]
+        this.nextProject = nextProject
+    }
+
+    /**
+     * Creates {@link Project} instances from a list of project arguments, chaining them and
+     * returning the first one to release.
+     *
+     * @param libero reference to the Libero object used to perform the actual release operations
+     * @param projectArguments list of strings containing the release information about the
+     *                         projects in the "<project>[:<releaseVersion>[:<nextVersion>]]" format
+     * @return reference to the first project to release
+     */
+    static Project createProjects(Libero libero, List<String> projectArguments) {
+        def nextProject = NO_OP_PROJECT
+
+        projectArguments.reverseEach {
+            projectArgument ->
+                nextProject = new Project(libero, projectArgument, nextProject)
+        }
+
+        return nextProject
+    }
+
+    /**
+     * Releases this project. If successful, the next project will automatically be released.
+     *
+     * @param options release options provided at the command line
+     */
+    void release(Options options) {
+        doRelease(options, getConfig())
+    }
+
+    /**
+     * Releases this project after a previous project has been successfully released. If successful,
+     * the next project will automatically be released.
+     *
+     * @param options release options provided at the command line
+     * @param previousProject project that was released before this one
+     */
+    protected void release(Options options, Project previousProject) {
+        doRelease(options, getConfig(previousProject))
+    }
+
+    protected Config loadConfig() {
+        def config = new Config();
+        config.configFile = new File("${projectName}.yml")
+        config.loadConfig()
+        return config
+    }
+
+    private doRelease(Options options, Config config) {
+        config.force = options.force
+        config.gitPush = options.gitPush
+        config.dryRun = options.dryRun
+        config.quickBuild = options.quickBuild
+
+        println "Releasing project ${projectName} ${releaseVersion} using the following configuration:"
+        config.printConfig()
+
+        libero.run(config)
+        nextProject.release(options, this)
+    }
+
+    private getConfig() {
+        def config = loadConfig()
+
+        config.projectName = projectName
```

Comment:
you shouldn't have to set this unless you want to use a specific name for the project. If it is not set Libero will default it to `new File(config.projectDir).getName()`